File: objc/TapstreamTest/CoreListenerImpl.m
#import "CoreListenerImpl.h"

@implementation CoreListenerImpl

@synthesize queue;

- (id)initWithQueue:(OperationQueue *)queueVal
{
	if((self = [super init]) != nil)
	{
		self.queue = queueVal;
	}
	return self;
}

- (void)reportOperation:(NSString *)op
{
	[queue put:AUTORELEASE([[Operation alloc] initWithName:op arg:nil])];
}

- (void)reportOperation:(NSString *)op arg:(NSString *)arg
{
	[queue put:AUTORELEASE([[Operation alloc] initWithName:op arg:arg])];
}

- (void)dealloc
{
	RELEASE(queue);
	SUPER_DEALLOC;
}

@end

File: cs/TapstreamTest/CoreListenerImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TapstreamMetrics.Sdk
{
    class CoreListenerImpl : CoreListener
    {
        public OperationQueue queue;

        public CoreListenerImpl(OperationQueue q)
        {
            queue = q;
        }

        public void ReportOperation(string op)
        {
            queue.Add(new Operation(op, null));
        }

        public void ReportOperation(string op, string arg)
        {
            queue.Add(new Operation(op, arg));
        }
    }
}
